src/test/util_string_tests.cpp

```diff
@@ -111,6 +125,15 @@ BOOST_AUTO_TEST_CASE(ConstevalFormatString_NumSpec)
     FailFmtWithError<2>("%1$*2$", err_term);
     FailFmtWithError<2>("%1$.*2$", err_term);
     FailFmtWithError<2>("%1$9.*2$", err_term);
+
+    // Ensure that tinyformat will throws if format string contains wrong number
+    // of specifiers. PassFmt relies on this to verify tinyformat successfully
+    // formats the strings, and will need to be updated if tinyformat is changed
+    // not to throw on failure.
+    BOOST_CHECK_EXCEPTION(TfmFormatZeroes<2>("%s"), tfm::format_error,
+        HasReason{"tinyformat: Not enough conversion specifiers in format string"});
```

Comment:
nit: don't feel strongly about it, just noting that `HasReason` can do partial matches, we can also check something shorter like:
```suggestion
        HasReason{"Not enough conversion specifiers"});
```
(which would enable this being a one-liner)

Reply:
Thanks for the suggestion, it's a trade-off, holding off on changing this.